stock_available_to_promise_release/models/stock_move.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2019 Camptocamp (https://www.camptocamp.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+from odoo.osv import expression
+from odoo.tools import float_compare
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    date_priority = fields.Datetime(
+        string="Priority Date",
+        index=True,
+        default=fields.Datetime.now,
+        help="Date/time used to sort moves to deliver first. "
+        "Used to calculate the ordered available to promise.",
+    )
+    ordered_available_to_promise = fields.Float(
+        "Ordered Available to Promise",
+        compute="_compute_ordered_available_to_promise",
+        digits=dp.get_precision("Product Unit of Measure"),
+        help="Available to Promise quantity minus quantities promised "
+        " to older promised operations.",
+    )
+    need_release = fields.Boolean()
+
+    @api.depends()
+    def _compute_ordered_available_to_promise(self):
+        for move in self:
+            move.ordered_available_to_promise = (
+                move._ordered_available_to_promise()
+            )
+
+    def _should_compute_ordered_available_to_promise(self):
+        return (
+            self.picking_code == "outgoing"
+            and not self.product_id.type == "consu"
+            and not self.location_id.should_bypass_reservation()
+        )
+
+    def _ordered_available_to_promise(self):
```

Comment:
Why is this method not part of _compute_ordered_available_to_promise ? Retrieving the value can be done by reading the field.

Reply:
As we can't have proper depends, if values change in the transaction, the field will not be recomputed.
We could add invalidations everywhere, not sure that's useful. The field is good for the view, but the code better has to recompute the value on every access. I should add a docstring about this.